File: frameworks/python/starlette/Implementation/routes/compressed.py
from starlette.middleware import Middleware
from starlette.middleware.gzip import GZipMiddleware
from starlette.requests import Request
from starlette.responses import Response
from starlette.routing import BaseRoute, Route

from payloads import Payloads
from serial import fresh

# rb:wiring compressed.*
# Starlette's GZipMiddleware on each of the family's routes, which gzips an answer when the request
# asks. A body under 500 bytes goes out as it is, which is the middleware's default, and gzip runs
# at level 1, the fastest level every framework here compresses at. Its own default is 9.
GZIP = [Middleware(GZipMiddleware, compresslevel=1)]
# rb:end


def routes(p: Payloads) -> list[BaseRoute]:
    """compressed: the payloads behind the route's own gzip middleware."""

    # rb:handler compressed.gzip_small,compressed.identity_small
    async def small(request: Request) -> Response:
        """
        responses:
          200: {description: items.small, content: {application/json: {schema: {$ref: "#/components/schemas/Payload"}}}}
        """
        return fresh(p.small)

    # rb:handler compressed.gzip_large,compressed.identity_large
    async def large(request: Request) -> Response:
        """
        responses:
          200: {description: items.large gzipped when the request asks, content: {application/json: {schema: {$ref: "#/components/schemas/Payload"}}}}
        """
        return fresh(p.large)

    return [
        Route("/compressed/small", small, middleware=GZIP),
        Route("/compressed/large", large, middleware=GZIP),
    ]

File: frameworks/java/ktor/Implementation/src/main/kotlin/implementation/routes/Compressed.kt
package implementation.routes

import implementation.Payloads
import implementation.fresh
import io.ktor.server.application.install
import io.ktor.server.plugins.compression.Compression
import io.ktor.server.response.respond
import io.ktor.server.routing.Route
import io.ktor.server.routing.get
import io.ktor.server.routing.route

/**
 * compressed: Ktor's Compression on these two routes alone, with its defaults: gzip and deflate for
 * a body of 200 bytes or more. Ktor compresses at the JDK Deflater's default level, 6, and has no
 * setting for it.
 */
fun Route.compressed(p: Payloads) {
    // rb:handler compressed.gzip_small,compressed.identity_small
    route("/compressed/small") {
        // rb:wiring compressed.*
        install(Compression)
        get { call.fresh(); call.respond(p.small) }
    }
    // rb:handler compressed.gzip_large,compressed.identity_large
    route("/compressed/large") {
        // rb:wiring compressed.*
        install(Compression)
        get { call.fresh(); call.respond(p.large) }
    }
}
